onnxruntime/core/platform/windows/env.cc

```diff
@@ -124,52 +147,159 @@ class WindowsEnv : public Env {
   }
 
   int GetNumCpuCores() const override {
-    SYSTEM_LOGICAL_PROCESSOR_INFORMATION buffer[256];
-    DWORD returnLength = sizeof(buffer);
-    if (GetLogicalProcessorInformation(buffer, &returnLength) == FALSE) {
-      // try GetSystemInfo
-      SYSTEM_INFO sysInfo;
-      GetSystemInfo(&sysInfo);
-      if (sysInfo.dwNumberOfProcessors <= 0) {
-        ORT_THROW("Fatal error: 0 count processors from GetSystemInfo");
+    if (core_vec_.empty()) {
+      return static_cast<int>(std::thread::hardware_concurrency());
```

Comment:
>std::thread::hardware_concurrency( [](http://example.com/codeflow?start=30&length=34)

For cores, I think we divide logical processor count by 2.

Reply:
Done, thanks!